prog/kubernetes/kubernetes_cluster_nexus.rb

```diff
@@ -294,6 +299,52 @@ def billing_rate_for(type, family)
     hop_wait
   end
 
+  # Dynamically injects static hosts DNS records for K8s nodes into CoreDNS Corefile after the kubernetes plugin block,
+  # then updates and replaces the kube-system/coredns ConfigMap to apply the changes.
+  label def sync_worker_dns_config
+    decr_sync_worker_dns_config
+
+    dns_records = kubernetes_cluster.all_functional_nodes.map { [it.vm.ip4.to_s, it.name].join(" ") }
+
+    coredns_configmap = YAML.load(kubernetes_cluster.client.kubectl("-n kube-system get cm coredns -oyaml"))
+    corefile = coredns_configmap["data"]["Corefile"]
+
+    # Remove existing Ubicloud hosts block if present (matches full block with comments)
+    # Example matched block (with ~8-space indent for contents):
+    #     hosts {
+    #         # Ubicloud Hosts
+    #         178.63.152.192 kcg0zevswgsp6hqepg9de83wwq-nz1df
+    #         178.63.152.197 knbvkn6c55gt1mfhab6znfv4j5-ukgtf
+    #         # End Ubicloud Hosts
+    #         fallthrough
+    #     }
+    corefile.gsub!(/^\s*hosts\s*\{\s*\n(?:\s*# Ubicloud Hosts\n)?(?:\s*        .*\n)*?\s*(?:# End Ubicloud Hosts\n)?\s*        fallthrough\s*\n\s*\}\s*$/m, "")
+    lines = corefile.split("\n")
+
+    # The new block should be right after the kuberentes block, so we aim to find the block and insert
+    # after the block ends
+    kubernetes_block_start = lines.find_index { |l| l.strip.end_with?("{") && l.include?("kubernetes") }
+    fail "Kubernetes block not found." unless kubernetes_block_start
+
+    end_idx = lines[(kubernetes_block_start + 1)..].find_index { |l| l.strip.end_with?("}") }
+    fail "Closing brace not found." unless end_idx
+    kubernetes_block_end = kubernetes_block_start + 1 + end_idx
+
+    # Build new hosts block with comment markers for future idempotency (split records for proper indentation)
+    hosts_lines = ["    hosts {"] +
+      ["        # Ubicloud Hosts"] +
+      dns_records.map { |l| "        #{l}" } +
+      ["        # End of Ubicloud Hosts",
+        "        fallthrough",
+        "    }"]
```

Comment:
It is hard to see visually that the correct indentation is used. You may want to consider an approach using `indent = " " * 4`, and doing `indent + "hosts {"`, `indent + indent + l`, and similar.

Reply:
You can also construct such multi-line strings using some heredoc for better readability.